src/NServiceBus.Persistence.DynamoDB.PersistenceTests/OutboxTests.cs

```diff
@@ -179,6 +180,37 @@ public async Task Should_delete_operations_when_set_as_dispatched()
         Assert.AreEqual(0, result.TransportOperations.Length);
     }
 
+    [Test]
+    public async Task Should_not_fail_when_dispatching_multiple_times()
+    {
+        var incomingMessageId = Guid.NewGuid().ToString();
+        ContextBag contextBag = configuration.GetContextBagForOutbox();
+
+        var transportOperations = new TransportOperation[24]; // 25 is the batch write limit
+        for (int i = 0; i < transportOperations.Length; i++)
+        {
+            transportOperations[i] = new TransportOperation(Guid.NewGuid().ToString(), new DispatchProperties(), ReadOnlyMemory<byte>.Empty, new Dictionary<string, string>());
+        }
+
+        using (var transaction = await configuration.OutboxStorage.BeginTransaction(contextBag))
+        {
+            var outboxMessage = new OutboxMessage(incomingMessageId, transportOperations);
+            await configuration.OutboxStorage.Store(outboxMessage, transaction, contextBag);
+
+            await transaction.Commit();
+        }
+
+        await configuration.OutboxStorage.SetAsDispatched(incomingMessageId, contextBag);
+
+        var secondAttemptContextBag = configuration.GetContextBagForOutbox();
+        var result1 = await configuration.OutboxStorage.Get(incomingMessageId, secondAttemptContextBag);
+        await configuration.OutboxStorage.SetAsDispatched(incomingMessageId, secondAttemptContextBag);
+
+        var result2 = await configuration.OutboxStorage.Get(incomingMessageId, configuration.GetContextBagForOutbox());
```

Comment:
what's the purpose of checking for a third time?

Reply:
Third?